applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -621,12 +620,14 @@ namespace Kratos
 
 		vector_diff = 0.50 * vector_diff / scalar_diff;
 
+
 		//rotate inital element basis
 		const double r0 = meanRotationScalar;
 		const double r1 = meanRotationVector[0];
 		const double r2 = meanRotationVector[1];
 		const double r3 = meanRotationVector[2];
 
+
 		Quaternion<double> q(r0, r1, r2, r3);
 		Vector rotatedNX0 = this->mNX0;

```

Comment:
Use reference, at least you need a copy and you don't update the value

Reply:
like Vector& rotatedNX0 ? if I do so my solutions are NaN

Reply:
I fixed the rest

Reply:
If Vector& rotatedNX0 is giving you NaN something not right is happening, right @RiccardoRossi ?

Reply:
so, how do we proceed?